Less configuration

surma opened the pull request:
I don't know if you want this, but I generalized your scripts so you don't have to change the paths in `build` and `start` for every new app.

I also enabled the user to write multiple executables (think cron jobs for examples) by adding subpackages. The `go` tool builds the whole app folder recursively.

Last but not least I made the `stop` script actually stop the program.

I have written a new [README](https://github.com/surma/openshift-go-diy/blob/bb6b2b3b8bc1d7cb697c41b6c4ddeec08358760f/README.md). If you're interested, let me know. I'll open a PR :)

surma committed:
Use `go get` recursively and less configuration

lkundrak commented:
Thanks for your pull request. I have some concerns that I'd like to be addressed before I pull the changeset:
- Your diff changes different things that the commit message says. Please split logical parts ("remove useless static parts", "move stop logic into stop hook", "run go get recursively", etc.) into separate commits and make sure the commit messages are up to date.
- You remove the example application (which is the point why this exists in the first place). Did you intend to move it and forgot to git add it?
- You remove $OPENSHIFT_INTERNAL_IP:8080 from the command line. That means an application would need to inspect environment itself. That is definitely a bad idea, since portability will suffer

Also, your README makes sense; but it sort of depends on getting this pulled, thus I can't integrate it now.

surma commented:
1. Agreed, will do. (I'll open a new, cleaned up PR once we get things figured out).
2. Yes, I removed the example application in favor of an abstract description in the `README` and because the directory containing the app code has to be named like the application on OpenShift. Since the app name will probably differ from user to user, an in-repo example application doesn't make sense this way.
3. I disagree. You basically enforce the program to take the address to listen on as a parameter. I enforce inspecting the environment. You can not always control how your program is called, you can always inspect the environment. I also prefer changing a few lines of checked and locally testable code than changing build scripts.

Awaiting your comments :)

lzap commented:
Guys, how far are you with this one? ;-)

fumin commented on .openshift/action_hooks/start:
```diff
@@ -4,8 +4,9 @@
-cat $OPENSHIFT_LOG_DIR/$OPENSHIFT_APP_NAME.log || :
-xargs -r kill <$OPENSHIFT_RUN_DIR/$OPENSHIFT_APP_NAME.pid || :
-nohup $OPENSHIFT_REPO_DIR/hello/hello $OPENSHIFT_INTERNAL_IP:8080 >$OPENSHIFT_LOG_DIR/$OPENSHIFT_APP_NAME.log 2>&1 </dev/null &
+nohup $OPENSHIFT_GEAR_NAME > $OPENSHIFT_LOG_DIR/$OPENSHIFT_APP_NAME.log 2>&1 </dev/null &
```
This will break for template files since the folder the executable resides is in "$GOPATH/bin" whereas the templates/static_files are still in "$OPENSHIFT_REPO_DIR/hello".
